contributors/design-proposals/node/troubleshoot-running-pods.md

```diff
@@ -106,243 +99,180 @@ subsequently be used to reattach and is reported by `kubectl describe`.
-`EphemeralContainer` type:
-
-```
-type EphemeralContainer struct {
-        metav1.TypeMeta
-        metav1.ObjectMeta
-
-        Spec EphemeralContainerSpec
-        Status v1.ContainerStatus
-}
-```
-
-`EphemeralContainerSpec` is similar to `v1.Container`, but contains only fields
-relevant to Debug Containers:
-
-```
-type EphemeralContainerSpec struct {
-        // Target is the pod in which to run the EphemeralContainer
-        // Required.
-        Target v1.ObjectReference
-
-        Name string
-        Image String
-        Command []string
-        Args []string
-        ImagePullPolicy PullPolicy
-        SecurityContext *SecurityContext
+type PodStatus struct {
+  ...
+  // Status for any Ephemeral Containers that running in this pod.
+  // This field is alpha-level and is only honored by servers that enable the EphemeralContainers feature.
+  // +optional
+  EphemeralContainerStatuses []ContainerStatus `json:"ephemeralContainerStatuses,omitempty" protobuf:"bytes,12,rep,name=ephemeralContainerStatuses"`
 }
 ```
 
-A new controller in the kubelet would watch for EphemeralContainers and
-create/delete debug containers. `EphemeralContainer.Status` would be updated by
-the kubelet at the same time it updates `ContainerStatus` for regular and init
-containers. Clients would create a new `EphemeralContainer` object, wait for it
-to be started and then attach using the pod's attach subresource and the name of
-the `EphemeralContainer`.
-
-Debugging is inherently imperative, however, rather than a state for Kubernetes
-to enforce. Once a Debug Container is started it should not be automatically
-restarted, for example. This solution imposes additionally complexity and
-dependencies on the kubelet, but it's not yet clear if the complexity is
-justified.
-
-### Debug Container Status
-
-The status of a Debug Container is reported in a new field in `v1.PodStatus`:
+`EphemeralContainerStatuses` resembles the existing `ContainerStatuses` and
+`InitContainerStatuses`, but `EphemeralContainers` introduces a new type:
 
 ```
-type PodStatus struct {
-        ...
-        EphemeralContainerStatuses []v1.ContainerStatus
+// An EphemeralContainer is a container which runs temporarily in a pod for human-initiated actions
+// such as troubleshooting. This is an alpha feature enabled by the EphemeralContainers feature flag.
+type EphemeralContainer struct {
+  // Spec describes the Ephemeral Container to be created.
+  Spec Container `json:"spec,omitempty" protobuf:"bytes,1,opt,name=spec"`
+
+  // If set, the name of the container from PodSpec that this ephemeral container targets.
+  // The ephemeral container will be run in the namespaces (IPC, PID, etc) of this container.
+  // If not set then the ephemeral container is run in whatever namespaces are shared
+  // for the pod.
+  // +optional
+  TargetContainerName string `json:"targetContainerName,omitempty" protobuf:"bytes,2,opt,name=targetContainerName"`
```

Comment:
all containers of the pods are in same network namespace except possibly pid namespace. Was this field introduced for the pid namespace case only ?